aioxmpp/avatar/service.py

```diff
@@ -401,35 +447,40 @@ def get_avatar_metadata(self, jid, *, require_fresh=False):
         ``feature-not-implemented`` or ``item-not-found`` and return
         an empty list of avatar descriptors, since this is
         semantically equivalent to not having an avatar.
-
         """
-        if require_fresh:
-            del self._metadata_cache[jid]
+        if not require_fresh:
+            try:
+                return self._metadata_cache[jid]
+            except KeyError:
+                pass
 
-        try:
-            return self._metadata_cache[jid]
-        except KeyError:
-            pass
+        with (yield from self._lock):
+            if jid in self._metadata_cache:
```

Comment:
I feel this can be written down nicer with try. Maybe:
```python
if require_fresh:
    self._metadata_cache.pop(jid, None)
else:
    try:
        return self._metadata_cache[jid]
    except KeyError:
        pass
```